src/contracts/Test/IbexaTestKernel.php

```diff
@@ -0,0 +1,184 @@
+        TransactionHandler::class => 'test.' . TransactionHandler::class,
+        Connection::class => 'test.doctrine.connection',
+        Repository\Repository::class => 'test.ibexa.repository',
+        Repository\ContentService::class => 'test.ibexa.content_service',
+        Repository\ContentTypeService::class => 'test.ibexa.content_type_service',
+        Repository\LanguageService::class => 'test.ibexa.language_service',
+        Repository\LocationService::class => 'test.ibexa.location_service',
+        Repository\ObjectStateService::class => 'test.ibexa.object_state_service',
+        Repository\PermissionResolver::class => 'test.ibexa.permission_resolver',
+        Repository\RoleService::class => 'test.ibexa.role_service',
+        Repository\SearchService::class => 'test.ibexa.search_service',
+        Repository\SectionService::class => 'test.ibexa.section_service',
+        Repository\UserService::class => 'test.ibexa.user_service',
+    ];
+
+    public function registerBundles()
+    {
+        return [
+            new SecurityBundle(),
+            new EzPublishCoreBundle(),
+            new EzPlatformCoreBundle(),
+            new EzPublishLegacySearchEngineBundle(),
+            new JMSTranslationBundle(),
+            new FOSJsRoutingBundle(),
+            new FrameworkBundle(),
+            new LiipImagineBundle(),
+            new TwigBundle(),
+            new DoctrineBundle(),
+        ];
+    }
+
+    public function registerContainerConfiguration(LoaderInterface $loader): void
+    {
+        $loader->load(static function (ContainerBuilder $container): void {
+            self::prepareEzPlatformFramework($container);
+            self::prepareDatabaseConnection($container);
+            self::createPublicAliasesForServicesUnderTest($container);
+            self::setUpTestLogger($container);
+        });
+    }
+
+    public function getCacheDir()
+    {
+        return sys_get_temp_dir() . '/ibexa-test-kernel/' . md5(get_class($this));
+    }
+
+    private static function prepareDatabaseConnection(ContainerBuilder $container): void
+    {
+        $container->loadFromExtension('doctrine', [
+            'dbal' => [
+                'memory' => true,
+                'driver' => 'pdo_sqlite',
+                'platform_service' => SqliteDbPlatform::class,
+                'logging' => false,
+            ],
+        ]);
+
+        $definition = new Definition(Repository\Tests\LegacySchemaImporter::class);
+        $definition->setPublic(true);
+        $definition->setArgument(0, new Reference('doctrine.dbal.default_connection'));
+        $container->setDefinition('test.ibexa.migrations.schema_importer', $definition);
+
+        $definition = new Definition(FixtureImporter::class);
+        $definition->setPublic(true);
+        $definition->setArgument(0, new Reference('doctrine.dbal.default_connection'));
+        $container->setDefinition('test.ibexa.migrations.fixture_importer', $definition);
```

Comment:
Is it possible to replace this with Yaml configuration?

Reply:
Replaced.